zerver/lib/events.py

```diff
@@ -1089,7 +1092,7 @@ def _draft_update_action(i: int) -> None:
             state["unsubscribed"] = [
                 stream
                 for stream in state["unsubscribed"]
-                if stream["stream_id"] not in deleted_stream_ids
+                if has_metadata_access_to_previously_subscribed_stream(user_profile, stream)
```

Comment:
Is this fine?

Reply:
Don't you also need to update the `archived` flag for streams that remain in this list, for example?

You should write a `test_events.py` test that verifies the case where an administrator unsubscribes from a stream and then archives it.

Reply:
Fixed.

Reply:
We no longer need to update the `archived` flag and `subscribers` list for streams that remain in this list as #30315 has been stacked into this PR and #30315 was responsible for fixing the incorrect stream logic where channels are incorrectly marked as `invite_only: true` and removed all the subscribers from channel when archiving.